web/Controllers/PropertyController.cs

```diff
@@ -12,52 +11,182 @@ namespace BricksAndHearts.Controllers;
 [Route("/property")]
 public class PropertyController : AbstractController
 {
-    private readonly IPropertyService _propertyService;
     private readonly IAzureMapsApiService _azureMapsApiService;
-    private readonly ILogger<PropertyController> _logger;
     private readonly IAzureStorage _azureStorage;
+    private readonly ILogger<PropertyController> _logger;
+    private readonly IPropertyService _propertyService;
 
-    public PropertyController(IPropertyService propertyService, IAzureMapsApiService azureMapsApiService, ILogger<PropertyController> logger, IAzureStorage azureStorage)
+    public PropertyController(IPropertyService propertyService, IAzureMapsApiService azureMapsApiService,
+        ILogger<PropertyController> logger, IAzureStorage azureStorage)
     {
         _propertyService = propertyService;
         _azureMapsApiService = azureMapsApiService;
         _logger = logger;
         _azureStorage = azureStorage;
     }
 
-    [Authorize(Roles = "Landlord")]
+    [HttpPost]
+    [Authorize(Roles = "Landlord, Admin")]
+    public ActionResult DeleteProperty(int propertyId)
+    {
+        var propDB = _propertyService.GetPropertyByPropertyId(propertyId);
+        if (propDB == null)
+        {
+            _logger.LogWarning("Property with ID {PropertyId} does not exist", propertyId);
+            return StatusCode(404);
+        }
+
+        if (GetCurrentUser().IsAdmin == false && GetCurrentUser().LandlordId != propDB.LandlordId)
+        {
+            _logger.LogWarning("You do not have access to any property with ID {PropertyId}.", propertyId);
+            return StatusCode(404);
+        }
+
+        var landlordId = propDB.Landlord.Id;
+
+        // Delete property
+        _propertyService.DeleteProperty(propDB);
+
+        // Go to View Properties page
+        return RedirectToAction("ViewProperties", "Landlord", new { id = landlordId });
+    }
+
+    [HttpGet]
+    [Route("/property/{propertyId:int}/view")]
+    public ActionResult ViewProperty(int propertyId)
+    {
+        var model = _propertyService.GetPropertyByPropertyId(propertyId);
+        if (model == null)
+        {
+            _logger.LogWarning("Property with ID {PropertyId} does not exist", propertyId);
+            return RedirectToAction("Error", "Home", new { status = 404 });
+        }
+
+        var propertyViewModel = PropertyViewModel.FromDbModel(model);
+        return View(propertyViewModel);
+    }
+
+    public PropertyViewModel UpdatePropertyViewModel(PropertyViewModel oldModel, PropertyViewModel updateModel)

```

Comment:
Is this used?

Reply:
Strange, it seems to not be used, and it also seems to be very similar to something in property service. I've removed it for now